### DISCUSSION TITLE
custom shell not updated in UI

### DISCUSSION THREAD

**Discussion by seatrout** [Taitava/obsidian-shellcommands#364] [Debugging and testing]
When a custom shell is set up for a command (in this instance WSL on a windows machine) the main UI screen still shows the default shell — see screenshot. 
![buglet](https://github.com/Taitava/obsidian-shellcommands/assets/5279653/d9df7e56-90d1-4100-98a0-26b22548efd8)

**Comment by Taitava** [comment]
Thank you for your report! Good catch! 😎 

I was able to reproduce this bug. I'll try to fix it in `0.20.1`.

Just for the record, as you probably already noticed, even though a wrong shell is shown in the settings view, executing the shell command does use the correct shell. I'm just mentioning this in case there are other people reading this. 🙂

**Reply by seatrout** [reply]
Yes. It does matter most that the right shell is executed! 

While we are at it, though, I just noticed an extra buglet: in the section underneath the window where the command is entered, the results of escaping it are displayed. And here, too, the display is not changed to reflect the (linux) shell actually in use. The special characters are still escaped with `backticks`, as for Windows and not `\` as they should be.

**Reply by Taitava** [reply]
So, if I understood correctly, you refer to the command preview text showing incorrect escape characters (backticks instead of backslashes)? Just like in the screenshot you posted in the first post?

The command preview (and escaping characters) are actually created by [a function that also produces the shell name](https://github.com/Taitava/obsidian-shellcommands/blob/655018eb0ab87955b2d5adf5adb2fda900431765/src/settings/setting_elements/CreateShellCommandFieldCore.ts#L121-L176) below the preview command, so fixing the shell name should also fix the escape characters. I just need to call the function again in a situation when the shell is changed. 👍

**Reply by seatrout** [reply]
Thanks. Yes. that will clear it up.

**Comment by Taitava** [comment]
# A fixed version `0.20.1` is now released
@seatrout can you test the new release, please? 🙂 If you still encouter any problems, please let me know. 👍

**Reply by seatrout** [reply]
Thank you. I will do so. In the meantime, my workaround has been to wrap complex sequences into a little PowerShell script and then run *that* from shell commands. All the quoting problems disappear!
